### DISCUSSION TITLE
Developing a CMS base project, need to dynamically append styles of an element.

### DISCUSSION THREAD

**Discussion by Mizan3050** [givanz/Vvveb#79] [General]
I am currently developing a CMS portal, the use case is very similiar to what vvveb does, i am stuck at a scenario, for example i am able to update the styles of a particular div, and I need to generate a different styles for the same div for mobile screen, i am following the same pattern of generating css selector hierarchy to set the style to an element as vvveb, but i am not able to retrive the existing style of the element and when i update it the existing other properites get overriden, for example 1st i set the font-size: 12px and next time i set the color: blue, the color is set but the font-size is gone, what do you think would be the right approach to acheive this, this is very properly done in vvveb, but i am unable to get how is it happening.

**Comment by givanz** [comment]
Saving everything inline with style attribute on the element will always overwrite any existing style as it has higher priority.

```js
node.style['font-size'] = '12px';
console.log(node.style['font-size']);
```

VvvebJs saves all properties as css in a style tag `#vvvebjs-styles` in the `<head>` of the page.

`<style id="vvvebjs-styles"></style>`

It uses `Vvveb.StyleManager` to handle reading and saving css properties when loading and saving component properties.

https://github.com/givanz/VvvebJs/blob/master/libs/builder/builder.js#L2441

`Vvveb.StyleManager` uses an internal object that has this format

```js
this.styles[media][selector][css_property][property_value];
this.styles["mobile"]["div#element"]["font-size"] = "12px";
```

On page load it parses the css from `#vvvebjs-styles` using `cssRules` https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/cssRules and populates the internal styles object.

https://github.com/givanz/VvvebJs/blob/master/libs/builder/builder.js#L2471-L2475

On property change it generates the css with proper `@media screen` for each media by iterating through the internal styles object 

https://github.com/givanz/VvvebJs/blob/master/libs/builder/builder.js#L2579

and saves it back to `#vvvebjs-styles`.

**Reply by Mizan3050** [reply]
Really appeciate your response, Thank you!!
